### PR TITLE
test improvements

### PR DESCRIPTION
**Pull request by grosser**
- always use bundler
 - make tests run on their own (no `-Itest -Ilib` required)
 - helper is not a test
 - replace distracting test runner with standard colors

screen shot 2017-04-10 at 10 52 57 am

### CODE REVIEW

**Comment by grosser** [comment]
@schneems

File: test/helper.rb [hunk 1]
```diff
@@ -1,40 +1,32 @@
 # Copyright (c) 2011 Evan Phoenix
 # Copyright (c) 2005 Zed A. Shaw
 
-begin
```
Comment (nateberkopec) [inline test/helper.rb]: I think this was added for a reason, I can't recall the issue.

File: test/test_app_status.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-require "test_helper"
```
Comment (nateberkopec) [inline test/test_app_status.rb]: What's wrong with test_helper? That's a pretty common convention.

File: test/helper.rb [hunk 3]
```diff
@@ -1,40 +1,32 @@
...
-  require "bundler/setup"
-rescue LoadError
-end
-
+require "bundler/setup"
 require "net/http"
 require "timeout"
 require "minitest/autorun"
-require "minitest/pride"
```
Comment (nateberkopec) [inline test/helper.rb]: No reason to remove IMO.

File: test/helper.rb [hunk 4]
```diff
@@ -1,40 +1,32 @@
...
-    if u.kind_of? String
-      res = Net::HTTP.get(URI.parse(u))
-    else
-      url = URI.parse(u[0])
-      res = Net::HTTP.new(url.host, url.port).start {|h| h.request(u[1]) }
-    end
-
-    assert res != nil, "Didn't get a response: #{u}"
-    results << res
+  uris.map do |u|
```
Comment (nateberkopec) [inline test/helper.rb]: fine with this

**Comment by grosser** [comment]
easier to see if it succeeded or failed ?
... I'd even prefer plain to pride since it's less distracting

**Comment by grosser** [comment]
it's not a test and `rake test` requires it ... which means it gets loaded
twice ... which is weird

**Comment by grosser** [comment]
let's kick it out and see what breaks then

**Comment by nateberkopec** [comment]
Re: bundler rescue load-error, I mean I think that was added as a request by debian (?) or some other package manager team that wants to run the tests w/o bundler.

**Comment by grosser** [comment]
kk, will add that as comment

**Comment by grosser** [comment]
@nateberkopec merge ?

File: gemfiles/2.1-Gemfile [hunk 5]
```diff
@@ -8,5 +8,6 @@ gem "rake-compiler"
 
 gem "rack", "~> 1.6"
 gem "minitest", '~> 5.9'
+gem "minitest-rb"
```
Comment (nateberkopec) [inline gemfiles/2.1-Gemfile]: Typo?

Comment (grosser) [inline gemfiles/2.1-Gemfile]: yep :D

Comment (grosser) [inline gemfiles/2.1-Gemfile]: ... should both use gemspec and gemspec should not be generated by hoe ...

Comment (grosser) [inline gemfiles/2.1-Gemfile]: fixed

**Comment by nateberkopec** [comment]
The only change here I'm really ok with is the one to `def hit` and the Bundler rescue comment.

* I don't want to change the color output, minitest-pride is fine. Sorry if you disagree.
* `test_helper` is a common idiom and doesn't need to be changed.

**Comment by grosser** [comment]
so given these 2 options, which you would chose the fabulously filed test ?

screen shot 2017-04-11 at 2 16 57 pm
screen shot 2017-04-11 at 2 16 33 pm

**Comment by grosser** [comment]
extracted bundler changes to it's own PR

**Comment by grosser** [comment]
so before I rebase this ...
 - you still like pride more than rg ?
 - is moving to helper ok ? (atm test_helper is treated as a test which is weird and could lead to load order issues)

**Comment by nateberkopec** [comment]
@grosser 

> you still like pride more than rg ?

Yes

> is moving to helper ok ? (

Alright, it's OK.

**Comment by grosser** [comment]
new clean PR instead of rebasing this mess :D 
https://github.com/puma/puma/pull/1283

**Pull request closed without merging**